qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -534,6 +504,9 @@ def lazy_setdefault(key):
         for gate_name, gate_num_qubits, params, equiv in rtn:
             logger.debug("%s/%s => %s\n%s", gate_name, gate_num_qubits, params, equiv)
 
+    # Remove dummy node in order to return graph to original state
+    graph.remove_node(dummy)
```

Comment:
We probably ought to ensure that this is a true finaliser on all the code that happens after `graph.add_node`, for example by doing
```python
dummy = graph.add_node(...)
try:
    graph.add_edges_from_no_data(...)
    rtn = ...
    ...
    return rtn
finally:
    graph.remove_node(dummy)
```

That makes us safe in the case that the function is exited because of some Python exception.

Reply:
Good. Will do.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/8790/commits/01a44e3d59db175b75b4f8b4511684fb56a14d65.